### ISSUE TITLE
Running OpenFunctions-v2 locally with llama.cpp

### ISSUE THREAD

**Issue by ChristianWeyer** [ShishirPatil/gorilla#247]
Did anyone ever manage to successfully run OpenFunctions-v2 locally - e.g. by using llama.cpp? Or a similar inference engine?

Thanks!

**Comment by knoopx** [comment]
no, just spits random choices, doesn't follow instructions and looks like their hosted endpoint uses openai underneath...

**Comment by tianjunz** [comment]
Hi Christian, really sorry to hear that! We have just added the scripts here: https://github.com/ShishirPatil/gorilla/tree/main/openfunctions. Could you let us know if this works or not? Happy to debug with you as well!

**Comment by CharlieJCJ** [comment]
Hi @ChristianWeyer! Thanks for your interest~

We provide GGUF quantized model and example inference walkthrough on https://huggingface.co/gorilla-llm/gorilla-openfunctions-v2-gguf if you are looking for doing inference on compressed model. We've tested them on consumer laptop!

We also provide example inference code for the original model, which you can check out this section in github repo https://github.com/ShishirPatil/gorilla/tree/main/openfunctions#end-to-end-example-1

Let us know if you encounter further issues! We're happy to provide support!

**Comment by ChristianWeyer** [comment]
Maybe, this is actually an issue with LiteLLM (https://github.com/BerriAI/litellm/issues/2209)
